template.c

```diff
@@ -0,0 +1,75 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* Expand template string using environment variables.
+ *
+ * If VAR1 is set to "abc" and VAR2 is not set, template
+ * "var1=${VAR1} var2=${VAR2} end" expands to
+ * "var1=abc var2= end".
+ */
+static void expand_template(char *tmpl, char *result)
+{
+	for (;;) {
+		char *start = strstr (tmpl, "${");
+		if (! start)
+			break;
+
+		char *end = strchr (start, '}');
+		if (! end)
+			break;
+
+		*start = '\0';
+		*end = '\0';
+
+		char const *envvar = getenv (start + 2);
+		if (! envvar)
+			envvar = "";
+
+		strcpy (result, tmpl);
+		result += strlen (tmpl);
+		sprintf (result, "%s", envvar);
+		result += strlen (envvar);
+
+		tmpl = end + 1;
+	}
+
+	/* Copy template part after the last variable. */
+	strcpy (result, tmpl);
+}
+
+int main (int argc, char **argv)
+{
+	char *tmpl = NULL;
+	char buffer[1024];
+	size_t n = 0;
```

Comment:
Nitpicking, but I think we could hardcode this in L67

Reply:
We cannot, because getline() takes pointer to size as a second argument.